### PR TITLE
Restructure map internals to pair field and dot

### PR DESCRIPTION
**Pull request by russelldb**
Many fields can be added/updated at the same time, so share a `dot`.
Using the `dot` as the key for merging therefore was incorrect.

See the test `dot_key_test` for the obvious counter example

### CODE REVIEW

**Comment by russelldb** [comment]
```
OK, passed 100000 tests

17.4770% {map_eqc,ctx_update,5}
17.4298% {map_eqc,update,4}
17.3969% {map_eqc,add,4}
13.6245% {map_eqc,replicate,3}
10.0407% {map_eqc,make_ring,2}
9.4975% {map_eqc,remove,2}
8.3714% {map_eqc,ctx_remove,4}
6.1623% {map_eqc,set_n,1}
true
```

Just FYI. The 1million run is still ongoing

File: src/riak_dt_map.erl [hunk 1]
```diff
@@ -690,6 +692,17 @@ get_for_key({_N, T}=K, ID, Dict) ->
 
 -endif.
 
+%% found by eqc. Using tag as key in merge left causes a bug like
```
Comment (cmeiklejohn) [inline src/riak_dt_map.erl]: Can you clean up this comment?  It's a bit hard to understand the way it's written now, and it appears to be one of the more important parts of this PR.

**Comment by russelldb** [comment]
Done!

**Comment by cmeiklejohn** [comment]
Great update on the comment, super easy to understand.  :+1: to merge!

**Comment by russelldb** [comment]
@cmeiklejohn This repo is Bors'd, so I think you need to +1 a hash.

**Comment by cmeiklejohn** [comment]
Can you please squash the commits so I can +1 that hash?

**Comment by cmeiklejohn** [comment]
:+1: 4ef2ab5

**Comment by russelldb** [comment]
I did, oh communication break down

**Comment by cmeiklejohn** [comment]
:+1: 5be9828d1a7e72d98381d6d0c7946105a7839402

**Comment by cmeiklejohn** [comment]
@borshop merge

**Pull request merged**